### PR TITLE
Profiles quick start guide

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Profiles quick start guide.

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/features/profiles.mdx [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+
+5. Check the `inputs.yaml` file as shown below (some lines are hidden). It outlines how to obtain the values for the model from the various tables. Rename the `identifies` table by replacing `rudder_events_production.web.identifies` with `profiles_sample.ecommerce.identifies`. Similarly, rename the `tracks` table by replacing `rudder_events_production.web.tracks` with `profiles_sample.ecommerce.tracks`. These names point to the test tables that you can access.
+
+<div class="infoBlock">
+Leave the rest of the file unchanged. However, if you want to add any new identifier for ID stitching or add any new tables to the project, you can do so here.
+</div>
+
+<img src="../assets/features/profiles-inputs-yaml.png" alt="Inputs yaml file" />
+
+6. Next, go to the `profiles.yaml` file where the models are defined. You will find two models here - `id_stitcher` and `feature_table_model`. For ID stitching, retain only the first model (`model_type: id_stitcher`) and **comment out the second model** (`model_type: feature_table_model`). Also, rename the `name` field from `domain_profile_id_stitcher` to `user_id_stitcher` as defined in **Step 4**.
```
Comment (dpatchigolla) [inline docs/features/profiles.mdx:127]: @soumyadebm This is a question to you: we have two models - `id_stitcher` and `feature_table_model`; Should we add a _model suffix to id_stitcher too? Or remove the suffix from `feature_table_model`? It felt inconsistent when I was writing this line in notion doc.

File: docs/features/profiles.mdx [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+<div class="infoBlock">
+Leave the rest of the file unchanged. However, if you want to add any new identifier for ID stitching or add any new tables to the project, you can do so here.
+</div>
+
+<img src="../assets/features/profiles-inputs-yaml.png" alt="Inputs yaml file" />
+
+6. Next, go to the `profiles.yaml` file where the models are defined. You will find two models here - `id_stitcher` and `feature_table_model`. For ID stitching, retain only the first model (`model_type: id_stitcher`) and **comment out the second model** (`model_type: feature_table_model`). Also, rename the `name` field from `domain_profile_id_stitcher` to `user_id_stitcher` as defined in **Step 4**. 
+
+<div class="infoBlock">
+It is important to rename the <code class="inline-code">name</code> field to the entity name set in <strong>Step 4</strong>. In doing so, you link the ID stitcher in the <code class="inline-code">pb_project.yaml</code> file to the model spec in <code class="inline-code">profiles.yaml</code>.
```
Comment (dpatchigolla) [inline docs/features/profiles.mdx:130]: > It is important to rename the

In the guide, we weren't asking to rename the entity name; So the `rename` may be confusing. Can you modify to just say `In the name field, it is important to use the same entity name defined in step 4`

File: docs/features/profiles.mdx [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+You can see that five distinct **MAIN_ID** are created, each representing an actual user. Also, **user4** and **user5** have the same **MAIN_ID** as they share the same email - highlighting successful ID stitching.
+
+### Part 3: Creating feature tables
+
+In this section, you will define the features mentioned in the [use case](#use-case). For this, you need to modify the `feature_table_model` model in the `profiles.yaml` file of your project folder.
+
+9. Uncomment the lines you commented out in <Link to="#part-2-setting-up-id-stitching">Step 6</Link> and modify the `name` from `domain_profile` to `user_profile`. Similar to `user_id_stitcher`, this is the name of the output table that will be created in the warehouse. Also, delete all the information present in the `vars` key - as you will <Link to="#feature-definitions">write your own features</Link> instead.
+
+<div class="infoBlock">
+You will write the features in the <code class="inline-code">entityvars</code> field. It is considered as a declarative form of SQL to define the features.
```
Comment (dpatchigolla) [inline docs/features/profiles.mdx:172]: > It is considered as a declarative form of SQL to define the features.

Can you rephrase to: `It can be considered as .....`

entityvars is a concept we came up with, so this makes more sense.

File: docs/features/profiles.mdx [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+2. Set up a connection profile to connect to your data warehouse. To do so, go to your terminal and enter `pb init profile` and follow the prompts. A sample profile for a Snowflake account is shown below:
+
+```
+Enter profile_name: test
+Enter target:  (default:dev):  # Just press enter, leaving it to default
+Enter account: vmskhef-customer_sample_data
+Enter warehouse: wh_guide
+Enter dbname: profiles_sample
+Enter schema: ecommerce_output
+Enter user: CUSTOMER_ADMIN
```
Comment (dpatchigolla) [inline docs/features/profiles.mdx:87]: @ameypv-rudder  Can you replace with following credentials:

Enter user: rudder_profiles
Enter password: pb360_b3ta!
Enter role: rudder_profiles_role

File: docs/features/profiles.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply docs/features/profiles.mdx:127]: in favor of adding a `model` suffix to `id_stitcher` for consistency - it is also quite explicit.

File: docs/features/profiles.mdx (hunk 3 above)
Comment (ameypv-rudder) [reply docs/features/profiles.mdx:172]: Sure - makes sense.

File: docs/features/profiles.mdx (hunk 2 above)
Comment (ameypv-rudder) [reply docs/features/profiles.mdx:130]: Sure - done!

**Review by dpatchigolla — APPROVED** [review]
Looks good.

File: docs/features/profiles.mdx [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+
+Companies gather user data across various digital touchpoints like websites, mobile apps, enterprise systems like CRMs, marketing platforms, etc. Users are also identified with multiple identifiers across their product journey, like their email ID, phone number, device ID, anonymous ID, account usernames, and more. In addition, they might have multiple identifiers as they often change their devices and use work and personal emails together.
+
+All of this data is typically scattered across multiple tables within a warehouse. **Profiles** is RudderStack's solution to tie all the user data from these tables into a unified view, giving you a 360-degree view of the user - often called **Customer 360**.
+
+<div class="infoBlock">
+Customer 360 is best represented as a table with one row per user, with multiple features or attributes computed for that user.
+</div>
+
+## 10-step guide to using Profiles
```
Comment (ishachopra0401) [inline docs/features/profiles.mdx:18]:
```suggestion
## Using Profiles
```

File: docs/features/profiles.mdx [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+This section is divided into four parts:
+
+- [Use case](#use-case)
+- [Setup](#part-1-setup)
+- [Setting up ID stitching](#part-2-setting-up-id-stitching)
+- [Create customer 360 feature tables](#part-3-creating-feature-tables)
+
+### Use case
+
+To understand how Profiles works, you will implement a use case by leveraging a sample dataset based on [RudderStack's ecommerce tracking plan](https://www.rudderstack.com/docs/event-spec/ecommerce-events-spec/). This data is available in Snowflake, which you can access by following the steps in this guide.
```
Comment (ishachopra0401) [inline docs/features/profiles.mdx:29]:
```suggestion
To understand how Profiles works, lets implement a use case by leveraging a sample dataset based on [RudderStack's ecommerce tracking plan](https://www.rudderstack.com/docs/event-spec/ecommerce-events-spec/). This data is available in Snowflake, which you can access by following the steps in this guide.
```

File: docs/features/profiles.mdx [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+
+You will create a simple Profiles project with the following capabilities:
```
Comment (ishachopra0401) [inline docs/features/profiles.mdx:31]:
```suggestion
You can create a simple Profiles project with the following capabilities:
```

File: docs/features/profiles.mdx [hunk 8]
```diff
@@ -0,0 +1,428 @@
...
+Enter password: pb360_b3ta!
+Enter role: rudder_profiles_role
+Append to /Users/<user_name>/.pb/siteconfig.yaml? Specify 'yes' to confirm, anything else to deny.
+yes
+```
+
+<div class="infoBlock">
+<ul>
+    <li>When prompted for a profile name in the CLI, enter <code class="inline-code">test</code>.</li>
+    <li>You will see the last line <strong>only if</strong> you already have set up Profiles earlier. Enter <code class="inline-code">yes</code> in this case.</li>
```
Comment (ishachopra0401) [inline docs/features/profiles.mdx:89]:
```suggestion
    <li>You will see the last line <strong>only if</strong> you have already set up Profiles earlier. Enter <code class="inline-code">yes</code> in this case.</li>
```

File: docs/features/profiles.mdx [hunk 9]
```diff
@@ -0,0 +1,428 @@
...
+</ul>
+</div>
+
+3. Run `pb init pb-project`. This command creates a sample project in a folder called `HelloPbProject`. To choose another name for the project folder like `ProfilesDemo`, run the following command:
+
+```python
+pb init pb-project -o ProfilesDemo
+```
+
+Your project folder will have the following structure. These files contain some default prefilled values that you need to modify.
```
Comment (ishachopra0401) [inline docs/features/profiles.mdx:99]:
```suggestion
Your project folder will have the following structure. These files contain some default prefilled values that you need to modify (as explained in the next section).
```

File: docs/features/profiles.mdx [hunk 10]
```diff
@@ -0,0 +1,428 @@
...
+```
+2023/01/27 16:19:24 Material Material_user_id_stitcher_27c35415_6 ran in 18.862061136s.
+``` 
+
+In the above example, the **table name** is `Material_user_id_stitcher_27c35415_6`, where:
+- `Material` is a common prefix.
+- `user_id_stitcher` is the name defined in **Step 4** and **Step 6**.
+- `27c35415_6` is a suffix generated by the Profile Builder tool. You can ignore this suffix for now.
+
+8. Go to the warehouse using the credentials defined in [Step 2](#part-1-setup) to see the actual table. Run the following query after replacing <table_name> with the table name from your CLI output:
```
Comment (ishachopra0401) [inline docs/features/profiles.mdx:154]: Should we remove the link from Step 2 as it becomes confusing when we say step 2 but it redirects to part 1

File: docs/features/profiles.mdx [hunk 11]
```diff
@@ -0,0 +1,428 @@
...
+
+You will see the following result:
+
+<img src="../assets/features/snowflake-table-query.png" alt="Snowflake table highlighting ID stitching" />
+
+You can see that five distinct **MAIN_ID** are created, each representing an actual user. Also, **user4** and **user5** have the same **MAIN_ID** as they share the same email - highlighting successful ID stitching.
+
+### Part 3: Creating feature tables
+
+9. To create a feature table, you need to modify the feature table model (`model_type: feature_table_model`) in `profiles.yaml`. Uncomment the lines you commented out in <Link to="#part-2-setting-up-id-stitching">Step 6</Link>. Then, modify the `name` from `domain_profile` to `user_profile`. Similar to `user_id_stitcher`, this is the name of the output table that is created in the warehouse. Also, delete all the information present in the `vars` key as you will <Link to="#feature-definitions">write your own features</Link> for the model instead.
```
Comment (ishachopra0401) [inline docs/features/profiles.mdx:168]: same as above.. the link on step 6 becomes confusing

File: docs/features/profiles.mdx [hunk 12]
```diff
@@ -0,0 +1,428 @@
...
+
+<div class="infoBlock">
+You will define the features mentioned in the <Link to="#use-case">use case</Link>. The <Link to="#feature-definitions">Feature definitions</Link> section contains details on each feature definition and the associated logic.
```
Comment (ishachopra0401) [inline docs/features/profiles.mdx:171]:
```suggestion
You can define the features mentioned in the <Link to="#use-case">use case</Link>. The <Link to="#feature-definitions">Feature definitions</Link> section contains details on each feature definition and the associated logic. 
```

File: docs/features/profiles.mdx [hunk 13]
```diff
@@ -0,0 +1,428 @@
...
+        - avg_session_length_in_sec_last_7_days
+```
+
+10. Go to your terminal and run the following command:
+
+```python
+pb run
+```
+
+In a few minutes, you will see a similar output as <Link to="#part-2-setting-up-id-stitching">Step 7</Link>. However, you will get two SQL files this time - one for the ID stitcher and feature table respectively. You will see also two tables created in the warehouse.
```
Comment (ishachopra0401) [inline docs/features/profiles.mdx:285]:
```suggestion
In a few minutes, you will see a similar output as <Link to="#part-2-setting-up-id-stitching">Step 7</Link>. However, you will get two SQL files this time - one for the ID stitcher and the other for the feature table. You will also see two tables created in the warehouse.
```

same as above for link on step 7

File: docs/features/profiles.mdx [hunk 14]
```diff
@@ -0,0 +1,428 @@
...
+		description: Optional field describing the feature.
+features:
+	- orders_completed_in_past_365_days
+```
+
+<div class="warningBlock">
+This feature would be a continuation to the <code class="inline-code">vars</code> block mentioned above; the <code class="inline-code">vars</code> key need not repeat again. The same applies to all the features described below.
+</div>
+
+- `first_seen_date`: Takes the earliest timestamp from both `tracks` and `identifies`. To do so, select both of these separately tables and then take the earliest value, as shown:
```
Comment (ishachopra0401) [inline docs/features/profiles.mdx:352]:
```suggestion
- `first_seen_date`: Takes the earliest timestamp from both `tracks` and `identifies`. To do so, select both of these tables separately and then take the earliest value, as shown:
```

File: docs/features/profiles.mdx [hunk 15]
```diff
@@ -0,0 +1,428 @@
...
+				order_by:
+					# In this example, we take the oldest occurence of the campaign source, so we sort the rows by timestamp in ascending order
+					- timestamp asc
+			from: inputs/rsIdentifies
+			where: context_campaign_source is not null and context_campaign_source != '' # We can filter rows with a where condition.  In this case, we ignore all rows where the campaign source is null or blank - to ensure we ignore organic visits
+features:
+  - campaign_source_first_touch  
+```
+
+- `avg_session_length_in_sec_last_7_days`: RudderStack captures a `session_id` and a `session_start_time` by default in the `page` and `track` calls. You can use this `session_id` to compute each session length and their average.
```
Comment (ishachopra0401) [inline docs/features/profiles.mdx:394]:
```suggestion
- `avg_session_length_in_sec_last_7_days`: RudderStack captures a `session_id` and a `session_start_time` by default in the `page` and `track` calls. You can use this `session_id` to compute each session's length and average. 
```

**Review by ishachopra0401 — APPROVED** [review]
